modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/DocumentMetadataSetDTOConverter.java

```diff
@@ -0,0 +1,183 @@
+import com.liferay.dynamic.data.mapping.model.DDMStructure;
+import com.liferay.dynamic.data.mapping.model.LocalizedValue;
+import com.liferay.dynamic.data.mapping.service.DDMStructureService;
+import com.liferay.headless.delivery.dto.v1_0.ContentStructureField;
+import com.liferay.headless.delivery.dto.v1_0.DocumentMetadataSet;
+import com.liferay.headless.delivery.dto.v1_0.Option;
+import com.liferay.headless.delivery.dto.v1_0.util.ContentStructureUtil;
+import com.liferay.petra.function.transform.TransformUtil;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.service.GroupLocalService;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.vulcan.dto.converter.DTOConverter;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterContext;
+import com.liferay.portal.vulcan.util.GroupUtil;
+import com.liferay.portal.vulcan.util.LocalizedMapUtil;
+
+import java.util.Locale;
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Sam Ziemer
+ */
+@Component(
+	property = "dto.class.name=com.liferay.dynamic.data.mapping.model.DDMStructure",
+	service = DTOConverter.class
+)
+public class DocumentMetadataSetDTOConverter
+	implements DTOConverter<DDMStructure, DocumentMetadataSet> {
+
+	@Override
+	public String getContentType() {
+		return DDMStructure.class.getSimpleName();
+	}
+
+	@Override
+	public DocumentMetadataSet toDTO(DTOConverterContext dtoConverterContext)
+		throws Exception {
+
+		DDMStructure ddmStructure = _ddmStructureService.getStructure(
+			(Long)dtoConverterContext.getId());
+
+		Group group = _groupLocalService.getGroup(ddmStructure.getGroupId());
+
+		return new DocumentMetadataSet() {
+			{
+				setActions(dtoConverterContext::getActions);
+				setAssetLibraryKey(() -> GroupUtil.getAssetLibraryKey(group));
+
+				setAvailableLanguages(
+					() -> LocaleUtil.toW3cLanguageIds(
+						ddmStructure.getAvailableLanguageIds()));
+
+				setContentStructureFields(
+					() -> TransformUtil.transformToArray(
+						ddmStructure.getRootFieldNames(),
+						fieldName -> _toContentStructureField(
+							dtoConverterContext.isAcceptAllLanguages(),
+							ddmStructure.getDDMFormField(fieldName),
+							dtoConverterContext.getLocale()),
+						ContentStructureField.class));
+				setDateCreated(ddmStructure::getCreateDate);
+				setDateModified(ddmStructure::getModifiedDate);
+
+				setDescription(
+					() -> ddmStructure.getDescription(
+						dtoConverterContext.getLocale()));
+
+				setDescription_i18n(
+					() -> LocalizedMapUtil.getI18nMap(
+						ddmStructure.getDescriptionMap()));
+				setId(ddmStructure::getStructureId);
+
+				setName(
+					() -> ddmStructure.getName(
+						dtoConverterContext.getLocale()));
+
+				setName_i18n(
+					() -> LocalizedMapUtil.getI18nMap(
+						ddmStructure.getNameMap()));
+				setSiteId(ddmStructure::getGroupId);
```

Comment:
Not saying it is wrong, but those blank lines between statements look weird. Did the SF ask you do that?

Reply:
I think you're right, it does look a bit odd, I've added a commit removing the extra lines. SF did not complain about the extra lines, but also does not complain when they are not there.